python/pyarts/workspace/workspace.py

```diff
@@ -143,11 +109,12 @@ def parse_function(func, allow_callbacks):
         arg_name = func_ast.args.args[0].arg
     except:
         raise Exception("Agenda definition needs workspace arguments.")
-
-    ws = Workspace(0)
-
+    
     context = copy(func.__globals__)
-    context.update({arg_name : ws})
+    assert isinstance(context.get(arg_name), Workspace), \
```

Comment:
Why is this required? Requiring the function argument to correspond to an existing variable is a pretty ugly clash with the Python syntax.

Reply:
I am also not really happy about this.

It's necessary to have a `Workspace` to create workspace variables and callback functions onto it.

You created a separate `InteractiveWorkspace` instance to deal with this problem.  I also need such an interface to the `Workspace` to get the workspace variables that are defined (and define new ones).

We have a mid-to-long term goal to make the `Workspace` own its entire context, so keeping your solution will not really be possible long-term.  I am therefore not really happy about your solution going forward.

Would you find it acceptable to have the arts_agenda take another input that contains the current workspace?  The check there would then rather read something like

```python
if global_workspace is None: 
    global_workspace = context.get(arg_name)
assert isinstance(global_workspace, Workspace), / ...
```

Then I would simply use `global_workspace` in the rest of the function.